morituri/program/flac.py

```diff
@@ -0,0 +1,19 @@
+from subprocess import check_call, CalledProcessError
+
+import logging
+logger = logging.getLogger(__name__)
+
+FLAC = 'flac'
+
+
+def encode(infile, outfile):
+    """
+    Encodes infile to outfile, with flac.
+    Uses '-f' because morituri already creates the file.
+    """
+    try:
+        check_call(['flac', '--totally-silent', '--verify', '-o', outfile,
+                    '-f', infile])
```

Comment:
> for i in $(ls *.flac); do flac -d $i -o foo.wav && flac --best --custom-option-1 foo.wav -o $1; done

^ Not safe. [[1]](http://mywiki.wooledge.org/ParsingLs#Enumerating_files_or_doing_stuff_with_files)

Rather use:
`for i in *.flac; do flac -d "$i" -o foo.wav && flac --best --custom-option-1 foo.wav -o "$1"; done`

Because, for example:
```
$ ls -1
flac file1.flac
flac file2.flac
other file.mp3

$ for i in $(ls *.flac); do echo $i ; done
flac
file1.flac
flac
file2.flac

# but:
$ for i in *.flac; do echo "$i" ; done
flac file1.flac
flac file2.flac
```

So maybe newcomers should not necessarily be expected to be proficient in something that even experienced users can slip sometimes...

[1] http://mywiki.wooledge.org/ParsingLs